pkg/table/tables/tables.go

```diff
@@ -82,6 +82,10 @@ type TableCommon struct {
 	// recordPrefix and indexPrefix are generated using physicalTableID.
 	recordPrefix kv.Key
 	indexPrefix  kv.Key
+
+	// avoid duplicated calculations of Cols
+	// mode -> cols
+	getColsCache [][]*table.Column
```

Comment:
Maybe we could use `hiddenColumns`, `visibleColumns` and `publicColumns` instead of this one.

Reply:
When are these fields supposed to be set (before this PR)? I don't find anywhere initializing them

Reply:
Seems never be setted. /cc @zimulala

Reply:
Seems https://github.com/pingcap/tidb/pull/50062 introduced the regression. It removed the default initialization, but mistakenly making them never initialized.